libraries/standard/mqtt/src/mqtt_lightweight.c

```diff
@@ -370,11 +373,31 @@ MQTTStatus_t MQTT_SerializePublishHeader( const MQTTPublishInfo_t * const pPubli
     return MQTTSuccess;
 }
 
-MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
+MQTTStatus_t MQTT_GetDisconnectPacketSize( size_t * pPacketSize )
 {
+    /* MQTT DISCONNECT packets always have the same size. */
+    *pPacketSize = MQTT_DISCONNECT_PACKET_SIZE;
+
     return MQTTSuccess;
 }
 
+MQTTStatus_t MQTT_SerializeDisconnect( const MQTTFixedBuffer_t * const pBuffer )
+{
+    MQTTStatus_t status = MQTTSuccess;
+
+    if( pBuffer->size < MQTT_DISCONNECT_PACKET_SIZE )
```

Comment:
We define an API wrapper for this value, above, but don't use it here.

Reply:
Changed to use GetDisconnectPacketSize.